src/Java.Interop.Tools.TypeNameMappings/Java.Interop.Tools.TypeNameMappings/JavaNativeTypeManager.cs

```diff
@@ -10,25 +10,33 @@
 using Mono.Cecil;
 using Java.Interop.Tools.Cecil;
 using Android.Runtime;
-#if !GENERATOR
 using Java.Interop.Tools.JavaCallableWrappers;
-#endif  // !GENERATOR
 #endif  // HAVE_CECIL
 
-namespace Java.Interop.Tools.TypeNameMappings {
+namespace Java.Interop.Tools.TypeNameMappings
+{
 
+#if HAVE_CECIL
+	public
+#endif
 	enum PackageNamingPolicy {
 		LowercaseHash,
 		Lowercase,
 		LowercaseWithAssemblyName,
 	}
 
-	public class JniTypeName
+#if HAVE_CECIL
+	public
+#endif
+	class JniTypeName
 	{
 		public string Type { get; internal set; }
 		public bool IsKeyword { get; internal set; }
 	}
 
+#if HAVE_CECIL
+	public
+#endif
 	static class JavaNativeTypeManager {
```

Comment:
I would prefer to not make this `public` until we've turned it into an "instance" class instead of a `static` class. Could that be done first/as part of this PR?

Reply:
Generally speaking "no", because the type is in Java.Interop.JavaCallableWrappers.dll and if that is public then the consumers (Xamarin.Android.Build.Tasks.dll, Xamarin.Android.JavadocImporter.dll, etc.) has no access to the this class.

It is NOT public in the mobile profile which is reasonable enough.